hbase-common/src/main/java/org/apache/hadoop/hbase/HConstants.java

```diff
@@ -1533,6 +1533,15 @@ public enum OperationStatusCode {
   // User defined Default TTL config key
   public static final String DEFAULT_SNAPSHOT_TTL_CONFIG_KEY = "hbase.master.snapshot.ttl";
 
+  // Soft drop for destructive table actions configuration
+  public static final String SNAPSHOT_BEFORE_DESTRUCTIVE_ACTION_ENABLED_KEY =
+    "hbase.snapshot.before.destructive.action.enabled";
+  public static final boolean DEFAULT_SNAPSHOT_BEFORE_DESTRUCTIVE_ACTION_ENABLED = false;
```

Comment:
I wish we could keep it ON by default

Reply:
I think this would be surprising for releasing branches, but we could do this for `master`